toolkit/tools/pkg/imagecustomizerlib/extractpartitions_test.go

```diff
@@ -127,3 +141,98 @@ func verifySkippableFrameMetadataFromFile(partitionFilepath string, magicNumber
 
 	return nil
 }
+
+// Tests partition extracting with partition resize enabled, but where the partition resize is a no-op.
+func TestCustomizeImageNopShrink(t *testing.T) {
+	var err error
+
+	baseImage := checkSkipForCustomizeImage(t, baseImageTypeCoreEfi)
+
+	buildDir := filepath.Join(tmpDir, "TestCustomizeImageNopShrink")
+	configFile := filepath.Join(testDir, "consume-space.yaml")
+	outImageFilePath := filepath.Join(buildDir, "image.qcow2")
+
+	// Customize image.
+	err = CustomizeImageWithConfigFile(buildDir, configFile, baseImage, nil, outImageFilePath, "", "raw-zst",
+		false /*useBaseImageRpmRepos*/, true /*enableShrinkFilesystems*/)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	espPartitionZstFilePath := filepath.Join(buildDir, "image_1.raw.zst")
+	rootfsPartitionZstFilePath := filepath.Join(buildDir, "image_2.raw.zst")
+
+	espPartitionFilePath := filepath.Join(buildDir, "image_1.raw")
+	rootfsPartitionFilePath := filepath.Join(buildDir, "image_2.raw")
+
+	// Check the file type of the output files.
+	checkFileType(t, espPartitionZstFilePath, "zst")
+	checkFileType(t, rootfsPartitionZstFilePath, "zst")
+
+	// Extract partitions.
+	err = extractZstFile(espPartitionZstFilePath, espPartitionFilePath)
+	if !assert.NoError(t, err) {
```

Comment:
very nit: assert.Error might be clearer than double negative !assert.NoError

Reply:
The `assert` functions affect the result of the test but don't automatically quit the test. This allows for multiple asserts to fail, instead of just one. But if an assert is fatal and will cause problems later on, you have to manually quit the test as shown here.